internal/it_automation/it_automation_default_policy_resource.go

```diff
@@ -0,0 +1,689 @@
+			},
+			"memory_allocation_unit": schema.StringAttribute{
+				Optional:    true,
+				Description: "Unit for memory allocation.",
+				Validators: []validator.String{
+					stringvalidator.OneOf("MB", "GB"),
+				},
+			},
+			"memory_pressure_level": schema.StringAttribute{
+				Optional:    true,
+				Description: "Sets memory pressure level to control system resource allocation during task execution.",
+				Validators: []validator.String{
+					stringvalidator.OneOf("Low", "Medium", "High"),
+				},
+			},
+		},
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *itAutomationDefaultPolicyResource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+	var plan itAutomationDefaultPolicyResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// find the default policy for this platform.
+	policy, diags := r.getDefaultPolicyByPlatform(ctx, plan.Platform.ValueString())
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// set initial state from discovered policy.
+	plan.ID = types.StringValue(*policy.ID)
+	plan.LastUpdated = types.StringValue(time.Now().Format(timeFormat))
+
+	// apply any configuration changes.
+	if r.hasConfigChanges(&plan) {
+		err := r.updateDefaultPolicyConfig(ctx, &plan)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Error updating default policy configuration",
+				err.Error(),
+			)
+			return
+		}
+
+		// get updated policy after configuration changes.
+		updatedPolicy, diags := getItAutomationPolicy(ctx, r.client, *policy.ID)
+		resp.Diagnostics.Append(diags...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+
+		plan.wrap(*updatedPolicy)
+	} else {
+		// no configuration changes, wrap with existing policy.
+		plan.wrap(*policy)
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (r *itAutomationDefaultPolicyResource) Read(
+	ctx context.Context,
+	req resource.ReadRequest,
+	resp *resource.ReadResponse,
+) {
+	var state itAutomationDefaultPolicyResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	policyID := state.ID.ValueString()
+	policy, diags := getItAutomationPolicy(ctx, r.client, policyID)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		// manually parse diagnostic errors.
+		// helper functions return standardized diagnostics for consistency.
+		// this is due to some IT Automation endpoints not returning structured/generic 404s.
+		for _, d := range resp.Diagnostics.Errors() {
+			if d.Summary() == policyNotFoundErrorSummary {
+				tflog.Warn(
+					ctx,
+					fmt.Sprintf(notFoundRemoving, fmt.Sprintf("IT Automation Default Policy %s", policyID)),
+				)
+				resp.State.RemoveResource(ctx)
+				return
+			}
+		}
+		return
+	}
```

Comment:
We append diags into resp before we check if there are any errors that should be ignored. This will result in terraform throwing an error even if the error is just a `policyNotFoundErrorSummary`

Something like this should allow us to return early if the error is a `policyNotFoundErrorSummary` else we append to `resp` and return the errors for terrform core to report.
```suggestion
	policy, diags := getItAutomationPolicy(ctx, r.client, policyID)

	if diags.HasError() {
		// manually parse diagnostic errors.
		// helper functions return standardized diagnostics for consistency.
		// this is due to some IT Automation endpoints not returning structured/generic 404s.
		for _, d := range resp.Diagnostics.Errors() {
			if d.Summary() == policyNotFoundErrorSummary {
				tflog.Warn(
					ctx,
					fmt.Sprintf(notFoundRemoving, fmt.Sprintf("IT Automation Default Policy %s", policyID)),
				)
				resp.State.RemoveResource(ctx)
				return
			}
		}

        resp.Diagnostics.Append(diags...)
		return
	}
```